WIP: feat(components): create reusable components FractionButtons

izmy opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

Resolve <!--- link the issue here -->

## Screenshots:

seibei-iguchi commented on packages/components/src/components/form/FractionButtons/FractionButtons.tsx:
```diff
@@ -0,0 +1,43 @@
+export type FractionButtonsProps = {
+    buttons: FractionButton[];
+};
+
```
We usually don't use this pattern, but rather subcomponents (like FractionButtons.Button). But in this case, I'd just export FractionButton, the container doesn't do anything special.

seibei-iguchi commented on packages/components/src/components/form/FractionButton/FractionButton.tsx:
```diff
@@ -0,0 +1,39 @@
+
+export type FractionButtonItem = {
+    label: string;
+    translation?: React.ReactNode;
```
I don't understand the need for both label and translation. You don't need a key at this level, I think. I'd just keep children prop.

seibei-iguchi commented on packages/components/src/components/form/FractionButton/FractionButton.tsx:
```diff
@@ -0,0 +1,39 @@
+    onClick: () => void;
+};
+
+export type FractionButtonProps = FractionButtonItem;
```
No need for these, just keep and export the props type.

seibei-iguchi commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeEthForm/Inputs.tsx:
```diff
@@ -71,42 +81,108 @@ export const Inputs = () => {
-                            <Translation id="TR_CLEAR_ALL" />
-                        </Button>
-                    )
+                options={[
```
I don't know if there's that much benefit in having different API for each of the props, I'd just make this ReactNodes, too. It seems confusing and we usually use the ReactNode approach anyway.

izmy replied:
The reason why I did props like this is I want to set a strict interface for component `InputWithOptions`, which basically renders options bellow the input and nothing else.

seibei-iguchi commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeEthForm/Inputs.tsx:
```diff
@@ -71,42 +81,108 @@ export const Inputs = () => {
+    );
+
     return (
-        <Column gap={spacings.sm} alignItems="center">
```
Why remove the center align?

izmy replied:
I had a problem with the width of the new component, but I fixed it, so I returned alignItems to its original state.

seibei-iguchi commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeEthForm/Inputs.tsx:
```diff
@@ -71,42 +81,108 @@ export const Inputs = () => {
-                        decimals={network.decimals}
+        <Column gap={spacings.sm}>
+            <InputWithOptions
+                input={
```
Do we need this? It seems it's always a NumberInput, since it has the fraction buttons.

tomasklim reviewed (changes requested):
Maybe it is just my feeling, but the fiat amount looks little bit more to the right than input. Maybe if you move it few pixels to the left, it would look better
![Screenshot 2024-12-28 at 9 24 43](https://github.com/user-attachments/assets/0356da1c-8e8d-4ecc-8f91-246893d53a22)

btw `Clear all` is missing in unstake
![Screenshot 2024-12-28 at 9 24 35](https://github.com/user-attachments/assets/b26d7ee4-2288-4564-b8ab-4490a81540e3)

`Max` in unstake is different amount than available amount
![Screenshot 2024-12-28 at 9 09 15](https://github.com/user-attachments/assets/fa31b538-c634-4ddc-8cb9-7a5d1ac96e8e)

Maybe it was there before but `Clear` button causes error state

https://github.com/user-attachments/assets/79a817a8-cf92-4af3-bc0e-e9d25e7b2657

tomasklim commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeEthForm/Inputs.tsx:
```diff
@@ -96,32 +96,34 @@ export const Inputs = () => {
         {
-            label: '10%',
-            tooltip,
+            label: '10 %',
```
Why space in `10 %`? In Unstake form there is no space. Please remove space

izmy replied:
Done... I also used translation keys for labels

tomasklim commented on packages/components/src/components/form/FractionButton/FractionButton.tsx:
```diff
@@ -0,0 +1,37 @@
+const DEFAULT_VARIANT = 'tertiary';
+
+export type FractionButtonProps = {
+    label: string;
```
let's remove label, I think it is not needed. Everything can be translatable. In English, it can be `10%` in different language it can be `10 %` or `1/10` in Korean. I dont know

tomasklim commented on packages/suite/src/views/wallet/coinmarket/common/CoinmarketForm/CoinmarketFormInputs.tsx:
```diff
@@ -19,20 +26,48 @@ import {
+): FractionButtonProps[] => {
+    return [
+        {
+            label: '1/4',
```
Use `%` please. I know there are fractions in current prod, but let's unify it.

Let's use same values from Stake from. Also change `All` to `Max`
<img width="467" alt="Screenshot 2025-01-04 at 11 20 26" src="https://github.com/user-attachments/assets/f560441f-93dd-4c94-9997-b6c10b7f988f" />

tomasklim commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeEthForm/Inputs.tsx:
```diff
@@ -71,28 +74,40 @@ export const Inputs = () => {
+                        : undefined
                 }
                 options={[
-                    {
```
please keep there `10%` so that it is unified with stake form

tomasklim reviewed (changes requested):
LGTM, few comments.
Could you please make `fiat` select narrower and maybe there should be no border?

<img width="189" alt="Screenshot 2025-01-04 at 11 23 26" src="https://github.com/user-attachments/assets/3776aa02-fe28-4607-b53a-d2347afdb967" />

It is hard to check commit by commit because the last two commits change logic from the first three commits. Maybe can you redo commits? Thanks

tomasklim commented on packages/suite/src/support/messages.ts:
```diff
@@ -189,10 +189,30 @@ export default defineMessages({
     },
+    TR_FRACTION_BUTTONS_10: {
+        defaultMessage: '10%',
+        id: 'TR_FRACTION_BUTTONS_10',
```
```suggestion
        id: 'TR_FRACTION_BUTTONS_10_PERCENT',
```
so it is clear

izmy replied:
done

tomasklim commented on packages/product-components/src/components/InputWithOptions/InputWithOptions.tsx:
```diff
@@ -0,0 +1,75 @@
+    fiatInputProps?: NumberInputProps<TFieldValues>;
+    fiatValue?: React.ReactNode;
+    options: FractionButtonProps[];
+    symbol: string;
```
would it make sense to call it `displaySymbol`? to prevent errors

izmy replied:
done

izmy commented:
@tomasklim 
- I added a small margin right to fiat amount
- Input now has a toggle between crypto and fiat value, so there is no **clear all** button
- I fixed the wrong **available amount**
- I still haven't worked on the **buy/sell form**, so I didn't change anything with selectbox, but I may create a follow-up issue for that

tomasklim commented on packages/suite/src/support/messages.ts:
```diff
@@ -189,10 +189,30 @@ export default defineMessages({
     TR_FRACTION_BUTTONS_ALL: {
         defaultMessage: 'All',
         id: 'TR_FRACTION_BUTTONS_ALL',
     },
```
remove and use just your new `TR_FRACTION_BUTTONS_MAX`

tomasklim commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeEthForm/Inputs.tsx:
```diff
@@ -49,38 +72,105 @@ export const Inputs = () => {
+                        ),
+                        variant: 'primary',
+                        onClick: () => {
+                            console.log('restakedReward', restakedReward);
```
console.log

tomasklim commented on packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeEthForm/Inputs.tsx:
```diff
@@ -74,55 +82,102 @@ export const Inputs = () => {
+    };
+
+    const tooltip = (
+        <Translation
+            id="TR_STAKE_MIN_AMOUNT_TOOLTIP"
+            values={{
+                amount: MIN_ETH_AMOUNT_FOR_STAKING.toString(),
+                symbol: displaySymbol,
+            }}
```
merge conflicts not resolved correctly, symbol -> networkDisplaySymbol
![Screenshot 2025-01-07 at 8 28 54](https://github.com/user-attachments/assets/9fa9cf89-3352-449d-9d1f-23112da3bd53)

tomasklim reviewed (changes requested):
add padding also to the top amount. Maybe 2px instead 4px everywhere would be enough?
![Screenshot 2025-01-07 at 8 30 26](https://github.com/user-attachments/assets/a1b71e63-e412-456b-a799-0d898bdafce6)

izmy commented:
> add padding also to the top amount. Maybe 2px instead 4px everywhere would be enough? ![Screenshot 2025-01-07 at 8 30 26](https://private-user-images.githubusercontent.com/33235762/400683670-a1b71e63-e412-456b-a799-0d898bdafce6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyNDYwMzAsIm5iZiI6MTczNjI0NTczMCwicGF0aCI6Ii8zMzIzNTc2Mi80MDA2ODM2NzAtYTFiNzFlNjMtZTQxMi00NTZiLWE3OTktMGQ4OThiZGFmY2U2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDEwMjg1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5Zjg4MDZlZDc3ZDZiYWRlOTRiZjE0OTE0M2MyNDY4ZWM0ZmJmNmFlN2I3Mjg3ZWI0ZDNkNGE1ZWE3NTU0MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DnhNd6AVHEVyEa2yLA48OIsOvgXTduk1EwuK9KOLBBQ)

After discussion, we decided to remove the padding.

seibei-iguchi commented on packages/components/src/components/form/FractionButton/FractionButton.tsx:
```diff
@@ -0,0 +1,37 @@
+
+export type FractionButtonProps = {
+    id: string;
+    translation: React.ReactNode;
```
I'd stick to children, to make it consistent with other components.

izmy replied:
Done.
